jumpscale/sals/zos/public_ip.py

```diff
@@ -0,0 +1,27 @@
+from jumpscale.core.exceptions import Input
+from jumpscale.clients.explorer.models import PublicIP, WorkloadType
+from typing import Union
+
+
+class PublicIPSGenerator:
+    """ """
+
+    def create(self, node_id: str, pool_id: int, ipaddress: str) -> PublicIP:
+        """Create a public IP reservation
+
+        Args:
+          node_id(str): id of the node where to reserve the PublicIP
+          pool_id(int) the capacity pool ID
+          ipaddress(str) the public IP that needs to be reserved.
+
+        Returns:
+          PublicIP: the newly created publicIP object
+        """
+
+        public_ip = PublicIP()
+        public_ip.ipaddress = ipaddress
```

Comment:
Here probably needs validation to match IPRange field used in PublicIP

Reply:
No I don't think so. You list the farmer's available IP's, then you pick one to create a publicIP type reservation. If you post incorrect data, the reservation will fail anyway.